src/main/scala/akka/stream/contrib/SortWithin.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.contrib
+
+import akka.stream.Attributes
+import akka.stream.impl.fusing.GraphStages.SimpleLinearGraphStage
+import akka.stream.stage.{GraphStageLogic, InHandler, OutHandler, TimerGraphStageLogic}
+
+import scala.collection.immutable
+import scala.concurrent.duration.FiniteDuration
+
+object SortWithin {
+
+  /**
+    * Factory for [[SortWithin]] instances.
+    *
+    * @param finiteDuration duration within which sorting needs to be done. Ideally should be kept as small as possible
+    * @tparam A input type
+    * @return [[SortWithin]] instance.
+    */
+  def apply[A <: Comparable[A]](finiteDuration: FiniteDuration): SortWithin[A] = new SortWithin[A](finiteDuration)
+
+  /**
+    * Java API: Factory for [[SortWithin]] instances.
```

Comment:
There should be documentation what SoprtWithin actually does, not that we're a factory here :)